### ISSUE TITLE
Improving guides with unit tests

### ISSUE THREAD

**Issue by danschultzer** [danschultzer/pow#230]
The following guides may be improved with unit test examples:

https://github.com/danschultzer/pow/blob/master/guides/custom_controllers.md
https://github.com/danschultzer/pow/blob/master/guides/lock_users.md
https://github.com/danschultzer/pow/blob/master/guides/sync_user.md
https://github.com/danschultzer/pow/blob/master/guides/user_roles.md

There's already unit test in this guide:

https://github.com/danschultzer/pow/blob/master/guides/redis_cache_store_backend.md

I'm not sure if it makes sense for all the above guides, but I think in the case of user roles it could definitely be useful.

**Comment by joshchernoff** [comment]
I just wanted to say that, the amount of effort going into guides on this project is really great thank you!

**Comment by danschultzer** [comment]
Thanks @joshchernoff!

Yeah, documentation is really important, and I feel it's too often neglected in open source libraries. It's understandable since you can only do so much with your free time, but I do agree with you that it's equally important if not more so.

There's also a lot of planned guides for the website: https://github.com/pow-auth/pow_site/issues

**Comment by cdvv7788** [comment]
@danschultzer https://github.com/danschultzer/pow/pull/496 Added a first example for unittests in the lock users guide.

**Comment by danschultzer** [comment]
Thanks! I've fleshed it out a bit more, but haven't tested it yet.

**Comment by danschultzer** [comment]
@cdvv7788 all tested, and pushed to the new https://github.com/pow-auth/pow_demo repo 😄

**Comment by cdvv7788** [comment]
@danschultzer nice! It was indeed a little problematic to test it. I had to create a project from scratch, configure it, etc. This should help with that in the future.
Question: how do you plan on maintaining the demo? Will we need to run through all of the branches and perform the updates in there? Does it make sense to have a `demo` generator (probably more work but should be easier to maintain)?

**Comment by danschultzer** [comment]
Yeah, I think a generator would be great to have! For now it's just rebase and force push, not great 😄 

Actually I've thought about a generator ever since I began working on Pow so you only had to run a single mix task without any adjustments. However there was no big libraries that did code injection, or at least I didn't find one that I particularly like.

Now with `phx.gen.auth` there might be a good way to do it: https://github.com/aaronrenner/phx_gen_auth/blob/master/lib/mix/phx_gen_auth/injector.ex
